ServiceBus/ServiceBus/Basket.cs

```diff
@@ -0,0 +1,32 @@
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated from a template.
+//
+//     Manual changes to this file may cause unexpected behavior in your application.
+//     Manual changes to this file will be overwritten if the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+namespace ServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    
+    public partial class Basket
+    {
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
+        public Basket()
+        {
+            this.BasketItems = new HashSet<BasketItem>();
+        }
+    
+        public int Id { get; set; }
+        public Nullable<int> CampaignId { get; set; }
+        public double TotalPrice { get; set; }
+    
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
+        public virtual ICollection<BasketItem> BasketItems { get; set; }
+        public virtual Campaign Campaign { get; set; }
```

Comment:
We should be able to use multiple campaigns in one basket (discounts could be combined - that is usually called combideal in e-comerce world).